backend/modules/simulation/agents/callbacks.py

```
@@ -41,18 +41,46 @@ def __init__(
+                    f"[PERSONA_CALLBACK] Error extracting response from LLMResult: {e}, "
+                    f"response type: {type(response)}, response keys: {dir(response) if hasattr(response, '__dict__') else 'N/A'}",
+                    exc_info=True
+                )
+                # Try to get response text from alternative locations
+                if hasattr(response, 'generations') and response.generations:
+                    if isinstance(response.generations[0], list) and len(response.generations[0]) > 0:
+                        if hasattr(response.generations[0][0], 'text'):
+                            response_text = response.generations[0][0].text
+                            self._log_conversation(response_text, processing_time)
+                        else:
+                            logger.error(f"[PERSONA_CALLBACK] Response object structure: {response.generations[0][0]}")
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)